elpis/gui/src/components/Engine/SelectEngine.js

```diff
@@ -1,52 +1,54 @@
-import React, { useEffect } from 'react';
+import React, { Component } from 'react';
 import { Grid, Segment, Header, Button, Dropdown, Divider } from 'semantic-ui-react';
 import { connect } from 'react-redux';
 import { translate } from 'react-i18next';
 import { engineList, engineLoad } from 'redux/actions/engineActions';
 import { setCurrentStep } from 'redux/actions/sideNavActions'
 
-const SelectEngine = props => {
-    let { t, currentEngine, list, engineList, engineLoad } = props;
+class SelectEngine extends Component {
```

Comment:
I changed this component to a class component as there were state problems the way it was previously. Prior to the change, reloading the app could trigger a circular update of state.